File: domain-service/credit-application/src/main/java/at/deloittedigital/core_banking/credit/datasource/model/Transaction.java
package at.deloittedigital.core_banking.credit.datasource.model;

import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Transaction {
    private String transactionId;
    private String accountId;
    private String accountIban;
    private String types;
    private String operation;
    private BigDecimal amount;
    private BigDecimal balance;
}

File: data-service/src/entity/Transaction.ts
import { ObjectType, Field } from "type-graphql";

@ObjectType()
export class Transaction {
  @Field()
  accountId: string;

  @Field()
  accountIban: string;

  @Field()
  type: string;

  @Field()
  amount: number;

  @Field()
  balance: number;

  @Field()
  transId: number;
}
